core/shared/src/main/scala/scalaz/effect/IO.scala

```diff
@@ -376,29 +376,48 @@ sealed abstract class IO[E, A] { self =>
   final def repeat[B](interval: Duration): IO[E, B] =
     self *> IO.sleep(interval) *> repeat(interval)
 
+  /**
+   * Repeats this action n time.
+   */
+  final def repeatN(n: Int): IO[E, A] = if (n <= 1) self else self *> repeatN(n - 1)
+
+  /**
+   * Repeats this action n time with a specified function, which computes
+   * a return value.
+   */
+  final def repeatNFold[B](n: Int)(b: B, f: (B, A) => B): IO[E, B] =
+    if (n < 1) IO.now(b) else self.flatMap(a => repeatNFold(n - 1)(f(b, a), f))
+
   /**
    * Repeats this action continuously until the first error, with the specified
    * interval between the start of each full execution. Note that if the
    * execution of this action takes longer than the specified interval, then the
    * action will instead execute as quickly as possible, but not
    * necessarily at the specified interval.
    */
-  final def repeatFixed[B](interval: Duration): IO[E, B] =
+  final def repeatFixed(interval: Duration): IO[E, A] =
     repeatFixed0(IO.sync(System.nanoTime()))(interval)
 
-  final def repeatFixed0[B](nanoTime: IO[Nothing, Long])(interval: Duration): IO[E, B] =
-    IO.flatten(nanoTime[E].flatMap { start =>
-      val gapNs = interval.toNanos
+  final def repeatFixed0(nanoTime: IO[Nothing, Long])(interval: Duration): IO[E, A] = {
```

Comment:
@jdegoes most of the changes are pure git-fu, either whole new files or trivial changes. This is the only thing i manually changed and can be lost within the big chunk of diffs.

Reply:
I've restored the type param